### PR TITLE
Split single- and multi-faceted VO definitions into their own functions

### LINKED ISSUE #5093: Split single- and multi-facet virtual object definition into separate API calls
## What is the Problem Being Solved?

Currently, `defineKind` and `defineDurableKind` each take a behavior parameter that specifies the shape of the VDO being define.  This object may be either a bag of method functions or a bag of facets, each of which is in turn a bag of method functions.  Several people have voiced concerns about the overloading of the behavior parameter makes type declarations very tricky and make the whole API harder to document and less clear in usage.

## Description of the Design

Instead, we have decided to break out the single- vs. multi-facet kind definitions into their own API entry points, so we will split `defineKind` into `defineKind` and `defineKindMulti`, and likewise split `defineDurableKind` into `defineDurableKind` and `defineDurableKindMulti`, where the first of each pair produces a single-facet VDO and expects `behavior` to _always_ be a bag of method functions, and the second of each pair produces a multi-facet VDO and expects `behavior` to _always_ be a bag of facets.

## Security Considerations

This is purely an ergonomic change and introduces no new security issues.

## Test Plan

Existing tests should be mostly sufficient, once updated to incorporate the single- vs. multi- distinction in the relevant places.  Possibly it might also be good to add a test of behavior parameter validation, e.g., to catch cases where older code in need of update passes a facet bag to `defineKind`.

### PR DESCRIPTION
**Pull request by FUDCo**
Closes #5093

### CODE REVIEW

File: packages/SwingSet/src/liveslots/virtualObjectManager.js [hunk 1]
```diff
@@ -156,8 +156,9 @@ export function makeCache(size, fetch, store) {
  * not need to occupy memory when they are not in use.  It provides five
  * functions:
  *
- * - `defineKind` and `defineDurableKind` enable users to define new types of
- *    virtual object by providing an implementation of the new kind of object's
+ * - `defineKind`, `defineKindMult`, `defineDurableKind`, and
```
Comment (turadg) [inline packages/SwingSet/src/liveslots/virtualObjectManager.js]:
```suggestion
 * - `defineKind`, `defineKindMulti`, `defineDurableKind`, and
```

File: packages/SwingSet/src/liveslots/virtualObjectManager.js [hunk 2]
```diff
@@ -564,11 +576,13 @@ export function makeVirtualObjectManager(
     const facetiousness = assessFacetiousness(behavior);
     switch (facetiousness) {
       case 'one': {
+        assert(!multifaceted);
```
Comment (turadg) [inline packages/SwingSet/src/liveslots/virtualObjectManager.js:579]: nice

File: packages/SwingSet/test/upgrade/test-upgrade.js [hunk 3]
```diff
@@ -74,11 +74,7 @@ async function testUpgrade(t, defaultManagerType) {
   const hostStorage = provideHostStorage();
   const { kvStore } = hostStorage;
   await initializeSwingset(config, [], hostStorage);
-  const c = await makeSwingsetController(
-    hostStorage,
```
Comment (turadg) [inline packages/SwingSet/test/upgrade/test-upgrade.js:78]: assuming this is misc cleanup

**Review by turadg — APPROVED** [review]
LGTM!

File: packages/SwingSet/test/upgrade/test-upgrade.js (hunk 3 above)
Comment (FUDCo) [reply packages/SwingSet/test/upgrade/test-upgrade.js:78]: Yes.  Debugging debris, presumably left over from @warner working on upgrade tests.

**Pull request merged**